src/modules/DNS.rb

```diff
@@ -246,24 +246,16 @@ def default_dhcp_hostname
     end
 
     def ReadHostname
-      fqhostname = ""
       # In installation (standard, or AutoYaST one), prefer /etc/install.inf
       # (because HOSTNAME comes with netcfg.rpm already, #144687)
       if (Mode.installation || Mode.autoinst) && FileUtils.Exists("/etc/install.inf")
         fqhostname = read_hostname_from_install_inf
       end
 
-      fqhostname = read_hostname_from_etc if fqhostname.empty?
+      # reads setup from /etc/HOSTNAME, returns a default if nothing found
+      fqhostname = Hostname.CurrentFQ if fqhostname.nil? || fqhostname.empty?
```

Comment:
I know. Updating yast2 is a next step. It has to be updated to fulfill bnc#923990. My primary target here was deduplicating the proposal.